From 2645df3771d7217551f5d0295a27a75dc62743a6 Mon Sep 17 00:00:00 2001
From: Anatoli Antonovitch <anatoli.antonovitch@amd.com>
Date: Mon, 29 May 2023 17:03:00 -0400
Subject: [PATCH 29/49] drm/amdgpu: Avoids registers access if device is
 physically gone

Clean up dmesg warning messages related to power subsystem for kobject 'i2c-0' when we
removed power for Navi23XM and detached the device from DomU:
sysfs group 'power' not found for kobject 'i2c-0';  sysfs group 'power' not found for kobject 'i2c-1'

Change-Id: Ib2a2359cf70457f0eee7e8a666a3c0b4554dc1a0
Signed-off-by: Anatoli Antonovitch <anatoli.antonovitch@amd.com>
---
 drivers/gpu/drm/amd/pm/swsmu/smu11/sienna_cichlid_ppt.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/amd/pm/swsmu/smu11/sienna_cichlid_ppt.c b/drivers/gpu/drm/amd/pm/swsmu/smu11/sienna_cichlid_ppt.c
index e2ad2b972ab0..07cf3a3ee80d 100644
--- a/drivers/gpu/drm/amd/pm/swsmu/smu11/sienna_cichlid_ppt.c
+++ b/drivers/gpu/drm/amd/pm/swsmu/smu11/sienna_cichlid_ppt.c
@@ -3894,6 +3894,10 @@ static void sienna_cichlid_i2c_control_fini(struct smu_context *smu)
 	struct amdgpu_device *adev = smu->adev;
 	int i;
 
+	/* Avoids registers access if device is physically gone */
+	if (!pci_device_is_present(adev->pdev))
+		return;
+
 	for (i = 0; i < MAX_SMU_I2C_BUSES; i++) {
 		struct amdgpu_smu_i2c_bus *smu_i2c = &adev->pm.smu_i2c[i];
 		struct i2c_adapter *control = &smu_i2c->adapter;
-- 
2.17.1

